CMakeLists.txt

```diff
@@ -505,6 +505,16 @@ if (LLAMA_VULKAN)
 
         add_compile_definitions(GGML_USE_VULKAN)
 
+
+        # Workaround to the iterator debug bug https://stackoverflow.com/questions/74748276/visual-studio-no-displays-the-correct-length-of-stdvector
+        if (MSVC)
+            if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+                add_compile_definitions(_ITERATOR_DEBUG_LEVEL=0)
+            else()
+                add_compile_options(/Zc:nrvo-)
+            endif()
+        endif()
```

Comment:
Why only do this when Vulkan is enabled? This seems like it should apply to all C++ code compiled by MSVC or clang-cl.

Reply:
I don't see this problem in other backends (i.e. cpu), and this is a workaround that causes a lost of debug information, so I don't want to fix something thats not break

Reply:
I am surprised that `/Zc:nrvo-` even has the intended effect because in all of the linked StackOverflow examples, the problematic vector is being returned (hence, Named Return Value Optimization applies) and the debugging is incorrect between the declaration and the time it is returned (because the storage is actually elsewhere).

The Kompute backend definitely [returns vectors](https://github.com/ggerganov/llama.cpp/blob/e23b974f4cf9270d05062d446f406e3ff55d9451/ggml-kompute.cpp#L167) and is affected by this, but I don't see where the Vulkan backend does.

Reply:
I agree that disabling NRVO doesn't make much sense here. @Adriankhl how can I reproduce this issue? I ran a test with a debug build with cl 19.29.30154 and I didn't get an errors.

Reply:
@slaren I did some MSVC update in the past few days and the error is now gone, and I failed to pinpoint the problematic version. Now I can only reproduce the issue by `clang-cl`:
```
cmake .. -GNinja -DCMAKE_C_COMPILER=clang-cl -DCMAKE_CXX_COMPILER=clang-cl -DCMAKE_EXPORT_COMPILE_COMMANDS=1 -DLLAMA_NATIVE=OFF -DLLAMA_VULKAN=ON -DCMAKE_BUILD_TYPE=Debug

ninja -j6

.\bin\embedding.exe -m ..\..\..\models\mxbai-embed-large-v1-f16.gguf -p "Antibiotics are a type of medication used to treat bacterial infections. They work by either killing the bacteria or preventing them from reproducing, allowing the body's immune system to fight off the infection. Antibiotics are usually taken orally in the form of pills, capsules, or liquid solutions, or sometimes administered intravenously. They are not effective against viral infections, and using them inappropriately can lead to antibiotic resistance.`nI love cat"
```

So I have updated my PR to only deal with `clang-cl`

Reply:
Thanks, I was able to reproduce this with `clang-cl`.